### ISSUE TITLE
ECR Repository Tags Support is Broken

### ISSUE THREAD

**Issue by eedwards-sk** [terraform-providers/terraform-provider-aws#8195]
### Overview

The API request for CreateRepository should include the tags, otherwise you cannot use IAM policies which restrict the API operation based on tags.

Current behavior does a "CreateRepository" API request, followed by a "TagResource" API request (see https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_ecr_repository.go#L60-L73)

This means that a proper locked down policy like this cannot work:

```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "CreateRepo",
      "Effect": "Allow",
      "Action": [
        "ecr:CreateRepository",
        "ecr:TagResource"
      ],
      "Resource": [
        "*"
      ],
      "Condition": {
        "StringEquals": {
          "aws:RequestTag/foo": "foo"
        },
        "ForAllValues:StringEquals": {
          "aws:TagKeys": [
            "foo"
          ]
        }
      }
    }
  ]
}
```

instead, you would have to use the following policy, which means the user ends up with permission to run CreateRepository on any kind of repository name / tag, which is a security risk and defeats the whole purpose of conditions...

```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "CreateRepo",
      "Effect": "Allow",
      "Action": [
        "ecr:CreateRepository"
      ],
      "Resource": [
        "*"
      ]
    },
    {
      "Sid": "SetRepoTags",
      "Effect": "Allow",
      "Action": [
        "ecr:TagResource"
      ],
      "Resource": [
        "*"
      ],
      "Condition": {
        "StringEquals": {
          "aws:RequestTag/foo": "foo"
        },
        "ForAllValues:StringEquals": {
          "aws:TagKeys": [
            "foo"
          ]
        }
      }
    }
  ]
}
```

### Terraform Version

0.11.13
provider: 2.4.0

### Affected Resource(s)

* aws_ecr_repository

### Terraform Configuration Files

```hcl
resource "aws_ecr_repository" "foo" {
  name = "foo"
  tags = {
    foo = "foo"
  }
}
```

### Debug Output

```
2019-04-04T20:02:21.247Z [DEBUG] plugin.terraform-provider-aws_v2.4.0_x4: 2019/04/04 20:02:21 [DEBUG] [aws-sdk-go] {"__type":"AccessDeniedException","Message":"User: <SNIPPED> is not authorized to perform: ecr:CreateRepository on resource: *"}
2019-04-04T20:02:21.247Z [DEBUG] plugin.terraform-provider-aws_v2.4.0_x4: 2019/04/04 20:02:21 [DEBUG] [aws-sdk-go] DEBUG: Validate Response ecr/CreateRepository failed, not retrying, error AccessDeniedException: User: <SNIPPED> is not authorized to perform: ecr:CreateRepository on resource: *
```

### Expected Behavior

The creation succeeds

### Actual Behavior

The creation fails due to permissions

### Steps to Reproduce

1. `terraform apply`

### References

* https://docs.aws.amazon.com/AmazonECR/latest/userguide/ecr-supported-iam-actions-tagging.html

**Comment by ewbankkit** [comment]
Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/6896

**Comment by bflad** [comment]
Support for tagging on create has been merged and will release with version 2.5.0 of the Terraform AWS Provider, likely later today. 👍

**Comment by eedwards-sk** [comment]
wow, I think I heard the sonic boom from how fast that turnaround was 💥 😎

**Comment by bflad** [comment]
This has been released in [version 2.5.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#250-april-05-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
